### DISCUSSION TITLE
CC Potentiometer Issues

### DISCUSSION THREAD

**Discussion by AtlasFatlas** [tttapa/Control-Surface#493] [Q&A]
Hi,
Got 7 Potentiometers (10K) wired in, and 1 of them behaves perfectly, but as soon as I physically connect more of them, they wig out, and I get erratic CC changes on every CC channel at once. I've tried using the array for multiple pots, and it still doesn't work. below is my code:

#include <Control_Surface.h> // Include the Control Surface library
 
// Instantiate a MIDI over USB interface.
USBMIDI_Interface midi;

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer1 = {
  A0,                                   // Analog pin connected to potentiometer
  {MIDI_CC::General_Purpose_Controller_1, CHANNEL_1}, // Channel volume of channel 1
};

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer2 = {
  A1,                                   // Analog pin connected to potentiometer
  {MIDI_CC::General_Purpose_Controller_2, CHANNEL_1}, // Channel volume of channel 1
};

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer3 = {
  A2,                                   // Analog pin connected to potentiometer
  {MIDI_CC::General_Purpose_Controller_3, CHANNEL_1}, // Channel volume of channel 1
};

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer4 = {
  A3,                                   // Analog pin connected to potentiometer
  {MIDI_CC::General_Purpose_Controller_4, CHANNEL_1}, // Channel volume of channel 1
};

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer5 = {
  A4,                                   // Analog pin connected to potentiometer
  {MIDI_CC::General_Purpose_Controller_5, CHANNEL_1}, // Channel volume of channel 1
};

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer6 = {
  A5,                                   // Analog pin connected to potentiometer
  {MIDI_CC::General_Purpose_Controller_6, CHANNEL_1}, // Channel volume of channel 1
};

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer7 = {
  A6,                                   // Analog pin connected to potentiometer
  {MIDI_CC::General_Purpose_Controller_7, CHANNEL_1}, // Channel volume of channel 1
};

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer8 = {
  A7,                                   // Analog pin connected to potentiometer
  {MIDI_CC::General_Purpose_Controller_8, CHANNEL_1}, // Channel volume of channel 1
};

// Instantiate a CCPotentiometer object
CCPotentiometer potentiometer9 = {
  A8,                                   // Analog pin connected to potentiometer
  {MIDI_CC::Channel_Volume, CHANNEL_1}, // Channel volume of channel 1
};

void setup() {
  Control_Surface.begin(); // Initialize Control Surface  
}
 
void loop() {
    Control_Surface.loop();
}

**Comment by AtlasFatlas** [comment]
Addition: I've tried multiple breadboards, different jumper leads, as well as different ground pins, I'm using an Arduino Due

**Comment by Veroledez** [comment]
Hola, prueba alimentado los potenciometros con una fuente externa y gnd comun,  o tambien prueba a conectar un pequeño capacitor  a las entradas analogicas, con valores por debajo de 10 uf

**Comment by dstoddart** [comment]
I had similar issues that were resolved when I added a 3.3v power supply to the due. It seems that all those pots create enough resistance to cause problems.
